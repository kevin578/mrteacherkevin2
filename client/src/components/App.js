import React, { Component } from "react";
import axios from "axios";
import Main from "./Main/Main";
import { BrowserRouter, Route } from "react-router-dom";
import { connect } from "react-redux";
import * as actions from "../actions";
import asyncComponent from "./AsyncComponent";


const Projects_async = asyncComponent(()=> import("./Projects/Projects"));
//Audacity
const Audacity1_async = asyncComponent(()=> import("./Pages/Audacity/Audacity1"));

//Arduino
const ArduinoBasics1_async = asyncComponent(()=> import("./Pages/ArduinoBasics/ArduinoBasics1"));
const ArduinoBasics2_async = asyncComponent(()=> import("./Pages/ArduinoBasics/ArduinoBasics2"));
const Arduino1_async = asyncComponent(()=> import("./Pages/Arduino/Arduino1"));
const Arduino2_async = asyncComponent(()=> import("./Pages/Arduino/Arduino2"));

//Google Chrome 
const GoogleApps1_async = asyncComponent(()=> import("./Pages/GoogleApps/GoogleApps1"));
const GoogleApps2_async = asyncComponent(()=> import("./Pages/GoogleApps/GoogleApps2"));


//Google Docs
const GoogleDocs1_async = asyncComponent(()=> import("./Pages/GoogleDocs/GoogleDocs1"));
const GoogleDocs2_async = asyncComponent(()=> import("./Pages/GoogleDocs/GoogleDocs2"));

//Google Sheets
const GoogleSheets1_async = asyncComponent(()=> import("./Pages/GoogleSheets/GoogleSheets1"));
const GoogleSheets2_async = asyncComponent(()=> import("./Pages/GoogleSheets/GoogleSheets2"));

//Google Slides
const GoogleSlides1_async = asyncComponent(()=> import("./Pages/GoogleSlides/GoogleSlides1"));
const GoogleSlides2_async = asyncComponent(()=> import("./Pages/GoogleSlides/GoogleSlides2"));

//HTML & CSS
const HtmlCss1_async = asyncComponent(()=> import("./Pages/HtmlCss/HtmlCss1"));

//Illustrator
const Illustrator1_async = asyncComponent(()=> import("./Pages/Illustrator/Illustrator1"));
const Illustrator2_async = asyncComponent(()=> import("./Pages/Illustrator/Illustrator2"));

//iMovie
const IMovie1_async = asyncComponent(()=> import("./Pages/IMovie/IMovie1"));
const IMovie2_async = asyncComponent(()=> import("./Pages/IMovie/IMovie2"));


//Lego Mindstorms
const LegoMindStorms1_async = asyncComponent(()=> import("./Pages/Engineering/LegoMindstorms1"));
const LegoMindStorms2_async = asyncComponent(()=> import("./Pages/Engineering/LegoMindstorms2"));

//LucidPress
const LucidPress1_async = asyncComponent(()=> import("./Pages/LucidPress/LucidPress1"));
const LucidPress2_async = asyncComponent(()=> import("./Pages/LucidPress/LucidPress2"));

//MacOS
const MacOS1_async = asyncComponent(()=> import("./Pages/MacOS/MacOS1"));
const MacOS2_async = asyncComponent(()=> import("./Pages/MacOS/MacOS2"));

//micro:bit
const Microbit1_async = asyncComponent(()=> import("./Pages/Microbit/Microbit1"));

//Photoshop
const Photoshop1_async = asyncComponent(()=> import("./Pages/Photoshop/Photoshop1"));
const Photoshop2_async = asyncComponent(()=> import("./Pages/Photoshop/Photoshop2"));

//Kerbal SpaceProgram
const KerbalSpaceProgram1_async = asyncComponent(()=> import("./Pages/Engineering/KerbalSpaceProgram1"));

//Scratch
const Scratch1_async = asyncComponent(()=> import("./Pages/Scratch/Scratch1"));
const Scratch2_async = asyncComponent(()=> import("./Pages/Scratch/Scratch2"));
const Scratch3_async = asyncComponent(()=> import("./Pages/Scratch/Scratch3"));

//Soundtrap
const Soundtrap1_async = asyncComponent(()=> import("./Pages/Soundtrap/Soundtrap1"));
const Soundtrap2_async = asyncComponent(()=> import("./Pages/Soundtrap/Soundtrap2"));

//Tinkercad
const Tinkercad1_async = asyncComponent(()=> import("./Pages/Tinkercad/Tinkercad1"));


//WeVideo
const WeVideo1_async = asyncComponent(()=> import("./Pages/WeVideo/WeVideo1"));
const WeVideo2_async = asyncComponent(()=> import("./Pages/WeVideo/WeVideo2"));



export function camelize(str) {
  return str
    .replace(/(?:^\w|[A-Z]|\b\w)/g, function(letter, index) {
      return index === 0 ? letter.toLowerCase() : letter.toUpperCase();
    })
    .replace(/\s+/g, "");
}

class App extends Component {
  componentDidMount() {
    this.getUserStateFromDatabase();
    this.getProjectInfoFromDatabase();
    this.props.fetchUser();
    this.props.setPage(0);
  }

  getUserStateFromDatabase = async () => {
    let state = await axios.get("/api/getStateFromDatabase");
    state = state.data;
    if (state) {
      this.props.completeInit(state.completed);
      if (state.score) this.props.changeScore(state.score);
      this.props.achievementInit(state.achievements);
    }
  };

  getProjectInfoFromDatabase = async ()=> {
    let projectData = await axios.get("/api/getUserProjects");
    projectData = projectData.data;
    if(projectData) this.props.setProjectsFromDatabase(projectData);

  }

  getRoutes() {
    return;
  }

  render() {
    return (
      <div>
        <BrowserRouter>
          <div>
            <Route path="/" component={Main} exact />
            <Route path = "/projects" component = {Projects_async}/>
            <Route path = "/[object Object]" component = {Main} />
            <Route path="/audacity1" component={Audacity1_async} />

            <Route path="/arduino1" component={Arduino1_async} />
            <Route path="/arduino2" component={Arduino2_async} />

            <Route path="/arduinobasics1" component={ArduinoBasics1_async} />
            <Route path="/arduinobasics2" component={ArduinoBasics2_async} />

            <Route path="/googleapps1" component={GoogleApps1_async} />
            <Route path="/googleapps2" component={GoogleApps2_async} />

            <Route path="/googledocs1" component={GoogleDocs1_async} />
            <Route path="/googledocs2" component={GoogleDocs2_async} />

            <Route path="/googlesheets1" component={GoogleSheets1_async} />
            <Route path="/googlesheets2" component={GoogleSheets2_async} />

            <Route path="/googleslides1" component={GoogleSlides1_async} />
            <Route path="/googleslides2" component={GoogleSlides2_async} />

            <Route path="/htmlcss1" component={HtmlCss1_async} />

            <Route path="/illustrator1" component={Illustrator1_async} />
            <Route path="/illustrator2" component={Illustrator2_async} />

            <Route path="/imovie1" component={IMovie1_async} />
            <Route path="/imovie2" component={IMovie2_async} />

            <Route path="/legomindstorms1" component={LegoMindStorms1_async} />
            <Route path="/legomindstorms2" component={LegoMindStorms2_async} />

            <Route path="/lucidpress1" component={LucidPress1_async} />
            <Route path="/lucidpress2" component={LucidPress2_async} />

            <Route path="/macos1" component={MacOS1_async} />
            <Route path="/macos2" component={MacOS2_async} />

            <Route path="/microbit1" component={Microbit1_async} />

            <Route path="/photoshop1" component={Photoshop1_async} />
            <Route path="/photoshop2" component={Photoshop2_async} />
            
            <Route
              path="/kerbalspaceprogram1"
              component={KerbalSpaceProgram1_async}
            />
            <Route path="/scratch1" component={Scratch1_async} />
            <Route path="/scratch2" component={Scratch2_async} />
            <Route path="/scratch3" component={Scratch3_async} />

            <Route path="/soundtrap1" component={Soundtrap1_async} />
            <Route path="/soundtrap2" component={Soundtrap2_async} />

            <Route path = "/tinkercad1" component = {Tinkercad1_async} />

            <Route path="/wevideo1" component={WeVideo1_async} />
            <Route path="/wevideo2" component={WeVideo2_async} />

          </div>
        </BrowserRouter>
      </div>
    );
  }
}

export default connect(
  null,
  actions
)(App);
