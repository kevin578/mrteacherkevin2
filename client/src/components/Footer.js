import React from "react";
import styled from "styled-components";

const Wrapper = styled.section`
  width: 100%;
  height: 200px;
  background: green;
`;

export const Footer = () => <Wrapper />;
