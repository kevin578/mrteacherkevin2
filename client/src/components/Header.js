import React from "react";
import styled from "styled-components";
import image from "./../img/logo2.png";
import { connect } from "react-redux";
import media from "./Main/mediaQueries";
import zeroFill from "zero-fill";
import PropTypes from "prop-types";
import googleButton from "../img/btn_google_light.svg";
import ReactSVG from "react-svg";
import hamburger from "../img/baseline-menu-24px.svg";

export const headerHeight = 60;
const imgHeight = headerHeight - 20;

const Wrapper = styled.section`
  top: 0px;
  background: blue;
  height: ${headerHeight}px;
  width: 100%;
  background: #345afb;
  display: flex;
  position: fixed;
  overflow: hidden;
  ${media.bigPhone`overflow: visible;`};
  z-index: 99;
`;

const Logo = styled.img`
  margin-top: ${(headerHeight - imgHeight) / 2}px;
  height: ${imgHeight}px;
  margin-left: 32px;
  cursor: pointer;
  ${media.bigPhone`  margin-left: 10px;`}
  ${media.smallPhone`height: 35px;`}
  ${media.smallPhone`margin-top: 12px;`}
`;

const Hamburger = styled(ReactSVG)`
  fill: white;
  margin-top: 7px;
  margin-left: 30px;
  display: none;
  ${media.bigPhone`display: block;`}
  ${media.smallPhone`margin-left: 10px;`}
`;

const HeaderLinks = styled.div`
  margin-left: 50%;
  display: flex;
  ${media.smallLaptop`margin-left: 40%;`};
  ${media.bigPhone`
  display: ${props=> props.show ? "block" : "none"};
  position: absolute;
  margin-top: 10px;
  left: 20px;
  width: 200px;
  top: 50px;
  margin-left: 0px;
  background: #fff;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  transition: all 0.3s cubic-bezier(.25,.8,.25,1);
  `};
`;

const HeaderLink = styled.a`
  margin-right: 40px;
  display: block;
  margin-top: ${headerHeight / 2}px;
  transform: translateY(-20%);
  color: #fff;
  font-weight: 600;
  cursor: pointer;
  text-decoration: none;
  ${media.bigPhone`
  color: #282828;
  width: 190px;
  border-bottom: 2px solid #cecece;
  padding-left: 10px;
  padding-bottom: 10px;
  `}

`;

const MobileHeaderLink = styled(HeaderLink)`
  display: none;
  ${media.bigPhone`display: block;`}
`;

const GoogleButtonLink = styled(HeaderLink)`
  margin-top: 9px;
  transform: translateY(0%);
  ${media.tablet`transform: translateX(-120px);`}
  ${media.bigPhone`display: none;`}
`;

export class Header extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      showHamburgerMenu: false
    };

  }

  getHeaderContent = () => {
    if (this.props.auth == null) {
      return;
    } else if (this.props.auth) {
      return (
        <HeaderLinks show = {this.state.showHamburgerMenu}>
          <HeaderLink>{zeroFill(5)(this.props.score)}</HeaderLink>
          <HeaderLink href="/api/logout">Logout</HeaderLink>
        </HeaderLinks>
      );
    } else {
      return (
        <HeaderLinks show = {this.state.showHamburgerMenu}>
          <GoogleButtonLink href = "/auth/google">
              <img src={googleButton} />
          </GoogleButtonLink>
          <MobileHeaderLink href="/auth/google">Sign in with Google</MobileHeaderLink>
        </HeaderLinks>
      );
    }
  };

  goHome = () => {
    window.location = "/";
  };

  hamburgerClicked = () => {
    this.setState((prevState)=> {
      return {
        showHamburgerMenu: !(prevState.showHamburgerMenu)
      };
    });
  }

  render() {
    return (
      <Wrapper>
        <Hamburger src={hamburger} onClick = {this.hamburgerClicked}/>
        <Logo src={image} onClick={this.goHome} />
        {this.getHeaderContent()}
      </Wrapper>
    );
  }
}

function mapStateToProps(state) {
  return {
    auth: state.auth,
    score: state.score
  };
}

export default connect(mapStateToProps)(Header);
