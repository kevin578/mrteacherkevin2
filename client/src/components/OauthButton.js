import React from 'react';
import styled from 'styled-components';



const ButtonBody = styled.button`
  font-family: 'Roboto', sans-serif;
  width: 200px;
  height: 50px;
  margin-left: auto;
  margin-right: auto;
  border: none;
  cursor: pointer;
  transition: 0.3s;
  border-radius: 3px;
  margin-top: 30px;
  background: white;
  display: flex;
  font-size: 14px;
  color: black;
  font-weight: 500;
`

const Logo = styled.img`
  height: 25px;
  position: relative;
  display: block;
  margin-top: 10px;
  transform: translateY();
  margin-right: 10px;
  margin-left: 3px;
`


const Text = styled.p`
  margin-top: 15px;
  margin-left: 6px;
  color: #5b5b5b;
`



const OauthButton = (props)=> {
    return (
        <ButtonBody>
                <Logo src = {props.logo} />
                <Text>{props.text}</Text>
        </ButtonBody>
    )
}

export default OauthButton;