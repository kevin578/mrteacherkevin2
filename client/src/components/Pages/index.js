
export { LegoMindstorms1 } from  "./Engineering/LegoMindstorms1";
//export * from  "./Engineering/LegoMindstorms1";
//export * from  "./Engineering/LegoMindstorms2";
