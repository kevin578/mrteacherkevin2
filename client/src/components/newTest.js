import React from 'react';
//import axios from 'axios'
import {connect} from 'react-redux';
import * as actions from '../actions';

class NewTest extends React.Component {
    componentDidMount() {
        axios.get('/api/getCoursePercentages')
            .then((resp)=> {
                console.log(resp);
            })
    }
    render() {
        return (
            <div>
                Look at this component
            </div>
        );
    }
}

function mapStateToProps(state) {


    return {
        auth: state.auth
    }
}

export default connect(null)(NewTest);