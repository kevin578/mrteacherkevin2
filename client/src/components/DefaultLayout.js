import React from "react";
import styled from "styled-components";
import Header from "./Header";
import Footer, { FOOTER_HEIGHT } from "./Footer/Footer";

const Page = styled.div`
  position: relative;
  min-height: 110vh;
`;

const Body = styled.div`
  margin-top: 85px;
  margin-left: 50px;
  padding-bottom: ${`${FOOTER_HEIGHT + 50}px`};

`;

const DefaultLayout = props => {
  return (
    <Page>
      <Header />
      <Body>{props.children}</Body>
      <Footer />
    </Page>
  );
};

export default DefaultLayout;