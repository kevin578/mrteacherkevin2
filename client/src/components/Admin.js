import React, { useState, useEffect } from 'react';
import DefaultLayout from './DefaultLayout';
import styled, {css} from 'styled-components';

const border = css`
  border: 1px solid #525252;
`;

const Table = styled.table`
  ${border}
  border-collapse: collapse;  
`;
const cell = styled.td`
  ${border}
  padding: 10px 20px;  
`;
const DateCell = styled(cell)``;
const CategoryCell = styled(cell)``;
const DescriptionCell = styled(cell)``;

const Admin = (props) => {
  const [recentActivity, setRecentActivity] = useState([]);
  const [visibleRecentActivity, setVisibleRecentActivity] = useState([]);

  useEffect(()=> {
    fetch('/api/getRecentActivity')
    .then(res => res.json())
    .then(res => {
      setRecentActivity(res);
    })
  }, []);

  function getDayName(date) {
    const days = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
    return days[date.getDay()];
  }

  function getTime(date) {
      const suffix = date.getHours() > 11 ? "PM" : "AM";
      const hour = date.getHours() > 12 ? date.getHours() - 12 : date.getHours();
      const minutes = date.getMinutes() > 9 ? date.getMinutes() : `0${date.getMinutes()}`
      return `${hour}:${minutes}${suffix}`
  }

  function renderRecentActivity() {
    return recentActivity.map((row)=> {
      const date = new Date(row.createdAt);

      return (
        <tr key = {row.id}>
          <DateCell>{`${getDayName(date)}, ${date.getMonth() + 1}/${date.getDate()}, ${getTime(date)}`}</DateCell>
          <CategoryCell>{row.category}</CategoryCell>
          <DescriptionCell>{row.description}</DescriptionCell>
        </tr>
      )
    })
  }

  return (
    <DefaultLayout>
      <Table>
        <tbody>
          <tr>
            <th>Time</th>
            <th>Type</th>
            <th>Description</th>
          </tr>
        { renderRecentActivity() }
        </tbody>
      </Table>
    </DefaultLayout>
  )
}

export default Admin;