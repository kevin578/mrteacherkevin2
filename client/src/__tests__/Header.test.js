import React from "react";
import styled from "styled-components";
import { shallow } from "../testConfig";
import  { Header } from "../components/Header";
import renderer from "react-test-renderer";

it("Header renders correctly", () => {
  const tree = shallow(<Header />);
  expect(tree).toMatchSnapshot();
});


