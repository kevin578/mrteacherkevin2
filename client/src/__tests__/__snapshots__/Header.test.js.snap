// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Unknown
          onClick={[Function]}
          src="baseline-menu-24px.svg"
/>,
        <Unknown
          onClick={[Function]}
          src="logo2.png"
/>,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
          "src": "baseline-menu-24px.svg",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
  fill: white;
  margin-top: 7px;
  margin-left: 30px;
  display: none;
  ",
              "
    @media (max-width: ",
              "38.75",
              "em) {
      ",
              "display: block;",
              ";
    }
  ",
              "
  ",
              "
    @media (max-width: ",
              "23.4375",
              "em) {
      ",
              "margin-left: 10px;",
              ";
    }
  ",
              "
",
            ],
          },
          "displayName": "Styled(ReactSVG)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
          "src": "logo2.png",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
  margin-top: ",
              "10",
              "px;
  height: ",
              "40",
              "px;
  margin-left: 32px;
  cursor: pointer;
  ",
              "
    @media (max-width: ",
              "38.75",
              "em) {
      ",
              "  margin-left: 10px;",
              ";
    }
  ",
              "
  ",
              "
    @media (max-width: ",
              "23.4375",
              "em) {
      ",
              "height: 35px;",
              ";
    }
  ",
              "
  ",
              "
    @media (max-width: ",
              "23.4375",
              "em) {
      ",
              "margin-top: 12px;",
              ";
    }
  ",
              "
",
            ],
          },
          "displayName": "styled.img",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "img",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      undefined,
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  top: 0px;
  background: blue;
  height: ",
          "60",
          "px;
  width: 100%;
  background: #345afb;
  display: flex;
  position: fixed;
  overflow: hidden;
  ",
          "
    @media (max-width: ",
          "38.75",
          "em) {
      ",
          "overflow: visible;",
          ";
    }
  ",
          ";
  z-index: 99;
",
        ],
      },
      "displayName": "styled.section",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "section",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Unknown
            onClick={[Function]}
            src="baseline-menu-24px.svg"
/>,
          <Unknown
            onClick={[Function]}
            src="logo2.png"
/>,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "src": "baseline-menu-24px.svg",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
  fill: white;
  margin-top: 7px;
  margin-left: 30px;
  display: none;
  ",
                "
    @media (max-width: ",
                "38.75",
                "em) {
      ",
                "display: block;",
                ";
    }
  ",
                "
  ",
                "
    @media (max-width: ",
                "23.4375",
                "em) {
      ",
                "margin-left: 10px;",
                ";
    }
  ",
                "
",
              ],
            },
            "displayName": "Styled(ReactSVG)",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "src": "logo2.png",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
  margin-top: ",
                "10",
                "px;
  height: ",
                "40",
                "px;
  margin-left: 32px;
  cursor: pointer;
  ",
                "
    @media (max-width: ",
                "38.75",
                "em) {
      ",
                "  margin-left: 10px;",
                ";
    }
  ",
                "
  ",
                "
    @media (max-width: ",
                "23.4375",
                "em) {
      ",
                "height: 35px;",
                ";
    }
  ",
                "
  ",
                "
    @media (max-width: ",
                "23.4375",
                "em) {
      ",
                "margin-top: 12px;",
                ";
    }
  ",
                "
",
              ],
            },
            "displayName": "styled.img",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "img",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        undefined,
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  top: 0px;
  background: blue;
  height: ",
            "60",
            "px;
  width: 100%;
  background: #345afb;
  display: flex;
  position: fixed;
  overflow: hidden;
  ",
            "
    @media (max-width: ",
            "38.75",
            "em) {
      ",
            "overflow: visible;",
            ";
    }
  ",
            ";
  z-index: 99;
",
          ],
        },
        "displayName": "styled.section",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "section",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
